backends/guest/common/util.c

```diff
@@ -45,17 +45,14 @@ void print_timestamp(timestamp_t t)
 	       t.second);
 }
 
-void read_timestamp(const uint8_t *esl_data)
+void read_timestamp(const struct guest_variable_header *data)
 {
 	timestamp_t timestamp;
 
-	if (esl_data == NULL)
+	if (data == NULL)
 		return;
 
-	// Special case: data read in from firmware contains a 16-bytes header, containing
-	//  a one-byte version number, then 15 bytes of timestamp -- truncating the trailing
-	//  padding byte at the end of the struct.
-	memcpy(&timestamp, esl_data + 1, TIMESTAMP_LEN - 1);
+	memcpy(&timestamp, &data->timestamp, sizeof(data->timestamp));
```

Comment:
> Why do we need a separate struct for the timestamp data?

We need a separate struct because timestamps are being used in 2 places:
1. When generating the auth file. This uses the full 16 byte timestamp_t structure. In the auth header is 16 bytes of timestamp, this value is also used in the hash. This auth structure we should try to keep consistent. These changes do not effect auth file generation.
2. When creating the exposed current secvar ESL. There is a 7 byte timestamp appended to the version right before the current esl. `libst-secvar` uses `unpack_signed_var` and `pack_signed_var` to cast between the 16 byte timestamp and 7 byte timestamp.

> Only changing the TIMESTAMP_LEN to 16 would suffice here.

If we read 16 bytes here then we would be loading in the first 9 bytes of esl data (9 bytes past the end of the timestamp) into the following `timestamp_t` values:
```
  uint8_t pad1;
  leint32_t nanosecond;
  leint16_t timezone;
  uint8_t daylight;
  uint8_t pad2;
  ```